wetterdienst/radolan.py

```diff
@@ -0,0 +1,136 @@
+""" Functions for DWD RADOLAN collection """
+from io import BytesIO
+from pathlib import Path
+from typing import Union, List, Tuple, Optional, Generator
+from datetime import datetime
+import logging
+
+from dateutil.parser import isoparse
+import pandas as pd
+
+from wetterdienst.additionals.functions import parse_enumeration_from_template
+from wetterdienst.constants.metadata import DWD_FOLDER_MAIN
+from wetterdienst.data_storing import store_radolan_data, restore_radolan_data
+from wetterdienst.download.download import download_radolan_data
+from wetterdienst.enumerations.column_names_enumeration import DWDMetaColumns
+from wetterdienst.enumerations.time_resolution_enumeration import TimeResolution
+from wetterdienst.file_path_handling.file_list_creation import create_filepath_for_radolan
+from wetterdienst.indexing.file_index_creation import create_file_index_for_radolan
+
+log = logging.getLogger(__name__)
+
+
+class DWDRadolanRequest:
+    """
+    API for DWD RADOLAN data requests
+    """
+    def __init__(self,
+                 time_resolution: Union[str, TimeResolution],
+                 date_times: Optional[Union[str, List[Union[str, datetime]]]] = None,
+                 start_date: Optional[Union[str, datetime]] = None,
+                 end_date: Optional[Union[str, datetime]] = None,
```

Comment:
By using Optional `= None` is redundant.

Reply:
type hint != default argument!